File: FCLLibrary/src/main/java/com/tungsten/fcllibrary/component/dialog/FCLDialog.java
package com.tungsten.fcllibrary.component.dialog;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatDialog;

import com.tungsten.fcllibrary.component.theme.ThemeEngine;

public class FCLDialog extends AppCompatDialog {

    public FCLDialog(@NonNull Context context) {
        super(context);
        ThemeEngine.getInstance().applyFullscreen(this, ThemeEngine.getInstance().getTheme().isFullscreen());
    }

}

File: FCLLibrary/src/main/java/com/tungsten/fcllibrary/component/dialog/FCLDialog.kt
package com.tungsten.fcllibrary.component.dialog

import android.content.Context
import androidx.appcompat.app.AppCompatDialog
import com.tungsten.fcllibrary.R
import com.tungsten.fcllibrary.component.theme.ThemeEngine

open class FCLDialog(context: Context) : AppCompatDialog(context) {
    init {
        ThemeEngine.getInstance()
            .applyFullscreen(window, ThemeEngine.getInstance().getTheme().isFullscreen)
        window?.setBackgroundDrawableResource(R.drawable.dialog_background)
    }
}
